src/test/regress/input/appendonly.source

```diff
@@ -811,3 +811,43 @@ insert into fix_ao_truncate_last_sequence select 1, 1 from generate_series(1, 5)
 select count(*) from fix_ao_truncate_last_sequence;
 abort;
 
+-- check reporting of missing attributes for AO table
+begin;
+create table test_reporting (id int) with (appendonly=true) distributed by (id);
+explain select xmin, * from test_reporting;
+abort;
```

Comment:
why do we need to explicitly open a transaction for each query and why create a new table for each query?

Reply:
Explicitly opened transaction allows to abort it and not deal with table cleanup, delete, etc.  New table for each query may be is not so necessary, but it brings up a test isolation idea. In any circumstances regardless of previous step outcome we will have a fresh table of required structure.

Reply:
the tests should be as simple as possible and in this case, the presence of an explicitly open transaction is not necessary.

Reply:
These tests are checking error processing and after error raised, the transaction switches into aborted state not allowing new statements inside it.  That's why transaction blocks are necessary here.